code/src/main/kotlin/com/expediagroup/sdk/v2/core/apache/util/HttpClientUtil.kt

```diff
@@ -1,18 +1,22 @@
 package com.expediagroup.sdk.v2.core.apache.util
 
-import com.expediagroup.sdk.v2.core.apache.interceptor.ExpediaGroupRequestInterceptor
-import com.expediagroup.sdk.v2.core.apache.interceptor.ExpediaGroupResponseInterceptor
-import com.expediagroup.sdk.v2.core.trait.common.IdTrait
 import com.expediagroup.sdk.v2.core.trait.configuration.ClientConfiguration
 import com.expediagroup.sdk.v2.core.trait.configuration.MaxConnectionsPerRouteTrait
 import com.expediagroup.sdk.v2.core.trait.configuration.MaxConnectionsTotalTrait
 import com.google.api.client.http.apache.v2.ApacheHttpTransport
 import org.apache.http.client.HttpClient
-import java.util.*
-import java.util.concurrent.ConcurrentHashMap
-import java.util.concurrent.ConcurrentMap
 
 
+/**
+ * Creates an instance of `HttpClient` based on the provided `ClientConfiguration`.
+ *
+ * This method requires that the `ClientConfiguration` implements both `MaxConnectionsTotalTrait`
+ * and `MaxConnectionsPerRouteTrait`.
+ *
+ * @param configuration The `ClientConfiguration` used to configure the `HttpClient` instance.
+ * @return A new instance of `HttpClient` configured according to the provided `ClientConfiguration`.
+ * @throws IllegalArgumentException If the provided `ClientConfiguration` does not implement the required traits.
+ */
```

Comment:
This function returns `HttpClient` imported from `org.apache.http.client`. I think it make since to move it to the file where we are building the transport, maybe combine the logic there to directly return the final `HttpClient` - unless there is a case where we need to get `ApacheHttpTransport` independently

Reply:
Done.